### PR TITLE
📝 Add Japanese translation for Tutorial - Security - First Steps

### PR DESCRIPTION
**Pull request by komtaki**
This PR translates the `tutorial/security/first-steps.md` to Japanese.

According to the issue: #1572

~~I merged the PR for dependency: https://github.com/tiangolo/fastapi/pull/1656~~
~~So, please check the commit : https://github.com/tiangolo/fastapi/commit/18f28de1e84cc3484a7258d738c1083363bccce2~~

Thanks!!

### CODE REVIEW

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    相対URLを使っているので、APIが`https://example.com/`にある場合、`https://example.com/token`を参照します。しかし、APIが`https://example.com/api/v1/`にある場合は`https://example.com/api/v1/token`を参照することになります。
+
+    相対 URL を使うことは、[プロキシと接続](./.../advanced/behind-a-proxy.md){.internal-link target=_blank}のような高度なユースケースでもアプリケーションが動作し続けることを確認するために重要です。
+
+このパラメーターはエンドポイント/ *path operation*を作成しません。しかし、URL`/token`はクライアントがトークンを取得するために使用するものであると宣言します。この情報は OpenAPI やインタラクティブな API ドキュメントシステムで使われます。
+
+実際のpath operationもすぐに作ります。
+
+!!! info "情報"
+    非常に厳格な「Pythonista」であれば、パラメーター名の`token_url`ではなく`tokenUrl`のスタイルが気に入らないかもしれません。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/security/first-steps.md:135]:
```suggestion
    非常に厳格な「Pythonista」であれば、パラメーター名のスタイルが`token_url`ではなく`tokenUrl`であることを気に入らないかもしれません。
```
Easier to read Japanese.

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+これで`oauth2_scheme`を`Depends`で依存関係に渡すことができます。
+
+```Python hl_lines="10"
+{!../../../docs_src/security/tutorial001.py!}
+```
+
+この依存関係は、*path operation function*のパラメーター`token`に代入される`str`を提供します。
+
+**FastAPI**は、この依存関係を使用してOpenAPIスキーマ(および自動APIドキュメント)で「セキュリティスキーム」を定義できることを知っています。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/security/first-steps.md:159]:
```suggestion
**FastAPI**は、この依存関係を使用してOpenAPIスキーマ (および自動APIドキュメント) で「セキュリティスキーム」を定義できることを知っています。
```

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+
+!!! info "技術詳細"
+    **FastAPI**は、クラス`OAuth2PasswordBearer` (依存関係で宣言されている) を使用してOpenAPIのセキュリティスキームを定義できることを知っています。これは`fastapi.security.oauth2.OAuth2`、`fastapi.security.base.SecurityBase`を継承しているからです。
+
+    OpenAPIと統合するセキュリティユーティリティ(および自動APIドキュメント)はすべて`SecurityBase`を継承しています。それにより、**FastAPI**はそれらをOpenAPIに統合する方法を知ることができます。
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/security/first-steps.md:164]:
```suggestion
    OpenAPIと統合するセキュリティユーティリティ (および自動APIドキュメント) はすべて`SecurityBase`を継承しています。それにより、**FastAPI**はそれらをOpenAPIに統合する方法を知ることができます。
```

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+
+## 機能
```
Comment (Attsun1031) [inline docs/ja/docs/tutorial/security/first-steps.md:166]:
```suggestion
## どのように動作するか
```

**Review by Attsun1031 — COMMENTED** [review]
Thank you for your tough work!
Please check comments I made.

**Comment by komtaki** [comment]
@Attsun1031 

Thank you for your review. 👍 
I applied your suggestion. So, check please.

**Review by Attsun1031 — APPROVED** [review]
LGTM 👍

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+
+そして、別のドメインか同じドメインの違うパス（またはモバイルアプリケーションの中）に **フロントエンド**を持っています。
+
+さらに、**ユーザー名**と**パスワード**を使って、フロントエンドがバックエンドと認証する方法を用意したいです。
+
+**FastAPI**では、これを**OAuth2**を使用して構築できます。
+
+ですが、ちょっとした必要な情報を探すために、長い仕様のすべてを読む必要はありません。
+
+**FastAPI**が提供するツールを使って、セキュリティを制御してみます。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:13]:
```suggestion
**FastAPI**が提供するツールを使って、セキュリティを制御してみましょう。
```

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+
+## どう見えるか
+
+まずはこのコードを使って、どう動くか観察します。その後で、何が起こっているのか理解するために戻ってきましょう。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:17]:
```suggestion
まずはこのコードを使って、どう動くか観察します。その後で、何が起こっているのか理解しましょう。
```

日本語ではこのぐらいシンプルな方が自然な気がします。

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 7]
```diff
@@ -0,0 +1,182 @@
...
+```
+
+## 実行
+
+!!! info "情報"
+    まず<a href="https://andrew-d.github.io/python-multipart/" class="external-link" target="_blank">`python-multipart`</a>をインストールします。
+
+    例えば、`pip install python-multipart`。
+
+    これは、**OAuth2**が「ユーザー名」や「パスワード」を送信するために、**フォームデータ** を使うからです。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:34]:
```suggestion
    これは、**OAuth2**が `ユーザー名` や `パスワード` を送信するために、「フォームデータ」を使うからです。
```

`hoge` はそのままで、"fuga" は「fuga」に変換するように統一されてきています。

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 8]
```diff
@@ -0,0 +1,182 @@
...
+$ uvicorn main:app --reload
+
+<span style="color: green;">INFO</span>:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
+```
+
+</div>
+
+## 確認
+
+インタラクティブなドキュメントにアクセスしてください: <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:50]:
```suggestion
次のインタラクティブなドキュメントにアクセスしてください: <a href="http://127.0.0.1:8000/docs" class="external-link" target="_blank">http://127.0.0.1:8000/docs</a>。
```

好みの問題だと思うので微妙なら無視してください。

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+    そして、あなたの*path operation*には、右上にクリックできる小さな鍵アイコンがあります。
+
+それをクリックすると、`ユーザー名`と`パスワード` (およびその他のオプションフィールド) を入力する小さな認証フォームが表示されます:
+
+<img src="/img/tutorial/security/image02.png">
+
+!!! note "備考"
+    フォームに何を入力しても、まだうまくいきません。ですが、これから動くようになります。
+
+もちろん最終的なユーザーのためのフロントエンドではありません。しかし、すべてのAPIをインタラクティブにドキュメント化するための素晴らしい自動ツールです。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:68]:
```suggestion
もちろんエンドユーザーのためのフロントエンドではありません。しかし、すべてのAPIをインタラクティブにドキュメント化するための素晴らしい自動ツールです。
```

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+
+これをフロントエンドチームが使用できます (また、あなた自身が使用することもできます) 。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:70]:
```suggestion
フロントエンドチームはこれを利用できます (また、あなたも利用できます) 。
```
簡潔に

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+* フロントエンドはAPIからさらにデータを取得する必要があります。
+    * しかし、特定のエンドポイントの認証が必要です。
+    * したがって、APIで認証するため、HTTPヘッダー`Authorization`に`Bearer`の文字列とトークンを加えた値を送信します。
+    * トークンに`foobar`が含まれている場合、`Authorization`ヘッダーの内容は次のようになります: `Bearer foobar`。
+
+## **FastAPI**の`OAuth2PasswordBearer`
+
+**FastAPI**は、これらのセキュリティ機能を実装するために、抽象度の異なる複数のツールを提供しています。
+
+この例では、**Bearer**トークンを使用して**OAuth2**を**パスワード**フローで使用します。これには`OAuth2PasswordBearer`クラスを使用します。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:106]: password flowの訳は「パスワードのフロー」か「パスワードフロー」のどちらかに統一したほうがいいと思います。後者の方が適切に感じます。

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 12]
```diff
@@ -0,0 +1,182 @@
...
+!!! info "情報"
+    「bearer」トークンが、唯一の選択肢ではありません。
+
+    しかし、私たちのユースケースには最適です。
+
+    あなたがOAuth2の専門家で、あなたのニーズに適した別のオプションがある理由を正確に知っている場合を除き、ほとんどのユースケースに最適かもしれません。
+
+    その場合、**FastAPI**はそれを構築するためのツールも提供します。
+
+クラス`OAuth2PasswordBearer`のインスタンスを作成する時に、パラメーター`tokenUrl`を渡します。このパラメーターには、クライアント (ユーザーのブラウザで動作するフロントエンド) がトークンを取得するために`ユーザー名`と`パスワード`を送信するURLを指定します。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:117]:
```suggestion
`OAuth2PasswordBearer` クラスのインスタンスを作成する時に、パラメーター`tokenUrl`を渡します。このパラメーターには、クライアント (ユーザーのブラウザで動作するフロントエンド) がトークンを取得するために`ユーザー名`と`パスワード`を送信するURLを指定します。
```

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+```Python hl_lines="6"
+{!../../../docs_src/security/tutorial001.py!}
+```
+
+!!! tip "豆知識"
+    ここで、`tokenUrl="token"`は、まだ作成していない相対URL`token`を指します。相対URLなので、`./token`と同じです。
+
+    相対URLを使っているので、APIが`https://example.com/`にある場合、`https://example.com/token`を参照します。しかし、APIが`https://example.com/api/v1/`にある場合は`https://example.com/api/v1/token`を参照することになります。
+
+    相対 URL を使うことは、[プロキシと接続](./.../advanced/behind-a-proxy.md){.internal-link target=_blank}のような高度なユースケースでもアプリケーションが動作し続けることを確認するために重要です。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:128]:
```suggestion
    相対 URL を使うことは、[プロキシと接続](./.../advanced/behind-a-proxy.md){.internal-link target=_blank}のような高度なユースケースでもアプリケーションを動作させ続けるために重要です。
```
簡潔に

File: docs/ja/docs/tutorial/security/first-steps.md [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+```Python hl_lines="10"
+{!../../../docs_src/security/tutorial001.py!}
+```
+
+この依存関係は、*path operation function*のパラメーター`token`に代入される`str`を提供します。
+
+**FastAPI**は、この依存関係を使用してOpenAPIスキーマ (および自動APIドキュメント) で「セキュリティスキーム」を定義できることを知っています。
+
+!!! info "技術詳細"
+    **FastAPI**は、クラス`OAuth2PasswordBearer` (依存関係で宣言されている) を使用してOpenAPIのセキュリティスキームを定義できることを知っています。これは`fastapi.security.oauth2.OAuth2`、`fastapi.security.base.SecurityBase`を継承しているからです。
```
Comment (tokusumi) [inline docs/ja/docs/tutorial/security/first-steps.md:162]:
```suggestion
    **FastAPI**は、`OAuth2PasswordBearer` クラス (依存関係で宣言されている) を使用してOpenAPIのセキュリティスキームを定義できることを知っています。これは`fastapi.security.oauth2.OAuth2`、`fastapi.security.base.SecurityBase`を継承しているからです。
```

**Review by tokusumi — CHANGES_REQUESTED** [review]
Thank you for tough work! I commented some improvements (there are a too tiny changes 😅 )

File: docs/ja/docs/tutorial/security/first-steps.md (hunk 11 above)
Comment (komtaki) [reply docs/ja/docs/tutorial/security/first-steps.md:106]: 統一しました 🙇

**Comment by komtaki** [comment]
@tokusumi 

Thanks for your review. I applied your suggestions and fixed. 👍 
So, check please.

**Review by tokusumi — APPROVED** [review]
LGTM!

**Comment by tiangolo** [comment]
Awesome, thank you @komtaki ! :cake: :coffee: 

And thanks @tokusumi and @Attsun1031 for the reviews! :bow:

**Pull request merged**